src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const length = this.length;
+
+      if (i === length - 1) {
+        result += this[i];
+        break;
+      }
+
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = '';
```

Comment:
Don't change the initial array. Now you change some element in `this` (in the initial array)

Reply:
Done)